Dockerfile

```diff
@@ -1,38 +1,52 @@
-FROM alpine:edge AS builder
+FROM phusion/baseimage:0.10.2 as builder
 LABEL maintainer="chevdor@gmail.com"
 LABEL description="This is the build stage for Substrate. Here we create the binary."
 
-RUN apk add build-base \
-    cmake \
-    linux-headers \
-    openssl-dev \
-    clang-dev \
-    cargo
-
 ARG PROFILE=release
 WORKDIR /substrate
 
 COPY . /substrate
 
-RUN cargo build --$PROFILE
+RUN apt-get update && \
+	apt-get upgrade -y && \
+	apt-get install -y cmake pkg-config libssl-dev git clang
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y && \
```

Comment:
We might want to download it ourselves, so that we can check the signatures.

Reply:
Can you provide details? I am not sure what you have in mind. You want to download Rust manually? Against what would you check the signatures?